platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesWriter.java

```diff
@@ -176,6 +188,32 @@ public void beginStreamingNewEvents(final @NonNull DoneStreamingPcesTrigger igno
         streamingNewEvents = true;
     }
 
+    /**
+     * Consider outstanding flush requests and perform a flush if needed.
+     *
+     * @return true if a flush was performed, otherwise false
+     */
+    private boolean processFlushRequests() {
+        boolean flushRequired = false;
+        while (!flushRequests.isEmpty() && flushRequests.peekFirst() <= lastWrittenEvent) {
+            final long flushRequest = flushRequests.removeFirst();
+
+            if (flushRequest > lastFlushedEvent) {
+                flushRequired = true;
+            }
+        }
+
+        if (flushRequired) {
+            if (currentMutableFile == null) {
+                logger.error(EXCEPTION.getMarker(), "Flush required, but no file is open. This should never happen");
+            }
+
+            closeFile();
```

Comment:
We don't need to close the file when we flush, just call the flush method on the stream object. As is, this will cause us to write one PCES file per round, which is not desirable.

Reply:
Fixed